src/utils/general_utils.py

```diff
@@ -56,3 +58,62 @@ def add_months_to_date(date_string, months):
         return result_date.strftime("%Y-%m-%d")
     except ValueError as e:
         raise ValueError("Invalid date format. Please use 'YYYY-MM-DD'.") from e
+
+
+# TODO: Might not scale well as we get more files in the blob
+def get_most_recent_date(mode, name_prefix, dataset):
+    """
+    Find files with the most recent date in their filename from Azure blob storage.
+
+    This function searches through Azure blob storage for files that start with the
+    given prefix and match the date pattern for the specified dataset. It returns
+    all files that match the most recent date found.
+
+    Parameters
+    ----------
+    mode : str
+        The mode in which the database is being accessed (e.g., 'local', 'dev').
+    name_prefix : str
+        The prefix of the filename before the date portion.
+        For example, 'seas5/monthly/processed/precip_em_i'.
+    dataset : str
+        Type of dataset. Must be one of: 'imerg', 'era5', 'seas5'.
+        This determines how the date is extracted from the filename.
+
+    Returns
+    -------
+    list of str
+        Names of all files that match the most recent date. Empty list if no
+        matching files are found.
+    """
+    container_client = get_container_client(mode, "raster")
+    blobs = container_client.list_blobs(name_starts_with=name_prefix)
+    file_dates = {}
+
+    for blob in blobs:
+        try:
+            date = parse_date(blob.name, dataset)
+            file_dates[blob.name] = date
+        except (ValueError, IndexError) as e:
+            print(f"Skipping {blob.name}: {str(e)}")
+            continue
+
+    if not file_dates:
+        return []
+
+    most_recent_date = max(file_dates.values())
+
+    return most_recent_date
+
+
+def parse_date(filename, dataset):
+    """
+    Parses the date based on a COG filename.
+    """
+    if (dataset == "era5") or (dataset == "imerg"):
+        date = pd.to_datetime(filename[-14:-4])
```

Comment:
Perhaps best to use a regex to pick up the dates here?

Reply:
Yes, a good point. I'll improve that